zerver/openapi/zulip.yaml

```diff
@@ -9234,6 +9234,18 @@ paths:
                           array will be empty if `enable_drafts_synchronization` is set to `false`.
                         items:
                           $ref: "#/components/schemas/Draft"
+                      emoji_data_url:
+                        type: string
+                        description: |
+                          The URL to a JSON file that describes which emoji names map to which
+                          emoji codes, for all Unicode emoji this Zulip server accepts.
+
+                          The data at the given URL is a JSON object with one property, `name_to_code`.
+                          The value of that property is a JSON object where each key is an
+                          [emoji name](/api/add-reaction#parameter-emoji_name) for an available
+                          Unicode emoji, and each value is the corresponding
+                          [emoji code](/api/add-reaction#parameter-emoji_code).
+                          A given emoji code may appear under multiple emoji names.
```

Comment:
This needs a `**Changes**` entry; search for that term to see a bunch of examples.

Reply:
We also might want to document any other special properties this JSON file will have. For example, to what extent will the URL be stable over time, and how should clients cache the file at this URL?

Reply:
The current plan (at https://github.com/zulip/zulip/pull/22067#discussion_r907836101 on the previous all-in-one PR, and in the chat thread linked there) is that (a) there are no guarantees about the URL changing or not changing, but (b) the HTTP response at that URL will have appropriate HTTP caching headers, so that your browser or other HTTP implementation should naturally DTRT.

It'd be good to mention briefly here the existence of those caching headers, for the sake of a reader wondering about this question.

(Also that part of the plan is the main thing I'm curious for empirical results on mobile about -- initial investigation is promising that the HTTP implementation React Native gives us should indeed make appropriate use of those headers, but we haven't yet actually tried that out ourselves.)